styx-api-service/src/test/java/com/spotify/styx/api/util/WorkflowFilteringTest.java

```diff
@@ -0,0 +1,214 @@
+import org.junit.Test;
+
+public class WorkflowFilteringTest {
+
+  public static final Instant TEST_DEPLOYMENT_TIME_BEFORE = Instant.ofEpochSecond(1638709400);
+  public static final Instant TEST_DEPLOYMENT_TIME_AFTER = Instant.ofEpochSecond(1638709300);
+
+
+  @Test
+  public void shouldReturnAllWorkflows() {
+    Collection<Workflow> workflowCollection = new ArrayList<>();
+    workflowCollection.add(Workflow.create(
+        "id-1", FLYTE_WORKFLOW_CONFIGURATION_WITH_DEPLOYMENT_SOURCE
+    ));
+
+    workflowCollection.add(Workflow.create(
+        "id-2", FLYTE_WORKFLOW_CONFIGURATION_WITH_DEPLOYMENT_SOURCE
+    ));
+
+    Map<FilterParams, String> emptyFilters = Map.of(DEPLOYMENT_TYPE, "",
+        DEPLOYMENT_TIME_BEFORE, "",
+        DEPLOYMENT_TIME_AFTER, "");
+
+    List<Workflow> workflows = filterWorkflows(workflowCollection, emptyFilters);
+
+    assertThat(workflows, equalTo(new ArrayList<>(workflowCollection)));
+  }
+
+  @Test
+  public void shouldNotReturnWorkflowsWithDeploymentType() {
+    Collection<Workflow> workflowCollection = new ArrayList<>();
+    workflowCollection.add(Workflow.create(
+        "id-1", FLYTE_WORKFLOW_CONFIGURATION_WITH_DEPLOYMENT_SOURCE
+    ));
+
+    workflowCollection.add(Workflow.create(
+        "id-2", FLYTE_WORKFLOW_CONFIGURATION_WITH_DEPLOYMENT_SOURCE
+    ));
+
+    Map<FilterParams, String> emptyFilters = Map.of(DEPLOYMENT_TYPE, "wrong-type",
+        DEPLOYMENT_TIME_BEFORE, "",
+        DEPLOYMENT_TIME_AFTER, "");
+
+    List<Workflow> workflows = filterWorkflows(workflowCollection, emptyFilters);
+
+    assertThat(workflows, empty());
+  }
+
+  @Test
+  public void shouldReturnWorkflowsWithDeploymentType() {
+    Collection<Workflow> workflowCollection = new ArrayList<>();
+    workflowCollection.add(Workflow.create(
+        "id-1", FLYTE_WORKFLOW_CONFIGURATION_WITH_DEPLOYMENT_TYPE
+    ));
+
+    Map<FilterParams, String> filters = Map.of(DEPLOYMENT_TYPE, "remote-foo",
+        DEPLOYMENT_TIME_BEFORE, "",
+        DEPLOYMENT_TIME_AFTER, "");
+
+    List<Workflow> workflows = filterWorkflows(workflowCollection, filters);
+
+    assertThat(workflows, equalTo(new ArrayList<>(workflowCollection)));
+  }
+
+  @Test
+  public void shouldReturnWorkflowsWithDeploymentTimeBefore() {
+    Collection<Workflow> workflowCollection = new ArrayList<>();
+    workflowCollection.add(Workflow.create(
+        "id-1", FLYTE_WORKFLOW_CONFIGURATION_WITH_DEPLOYMENT_TYPE
+    ));
+
+    Map<FilterParams, String> filters = Map.of(DEPLOYMENT_TYPE, "remote-foo",
+        DEPLOYMENT_TIME_BEFORE, TEST_DEPLOYMENT_TIME_BEFORE.toString(),
+        DEPLOYMENT_TIME_AFTER, "");
+
+    List<Workflow> workflows = filterWorkflows(workflowCollection, filters);
+
+    assertThat(workflows, equalTo(new ArrayList<>(workflowCollection)));
+  }
+
+  @Test
+  public void shouldReturnWorkflowsWithDeploymentTimeAfter() {
+    Collection<Workflow> workflowCollection = new ArrayList<>();
+    workflowCollection.add(Workflow.create(
```

Comment:
Same regarding having at least one more workflow in the collection with deployment time NOT after